src/app/category/category.component.html

```diff
@@ -0,0 +1,45 @@
+<p-card>
+  <div class="p-fluid grid">
+    <div class="field col-12 md:col-10">
+      <input type="text" pInputText />
+    </div>
+    <div class="field col-12 md:col-2">
+      <p-splitButton label="Add" icon="pi pi-plus"></p-splitButton>
+    </div>
+  </div>
+  <p-table
+    [value]="category"
+    styleClass="p-datatable-gridlines"
+    responsiveLayout="scroll"
+  >
+    <!-- <ng-template pTemplate="caption"> Header </ng-template> -->
+
+    <ng-template pTemplate="header">
+      <tr>
+        <th>Name</th>
+        <th></th>
+      </tr>
+    </ng-template>
+    <ng-template pTemplate="body" let-category>
```

Comment:
you have the category stored in a variable _category_ , that's what you should use not _category